codex/sales/reservations.nim

```diff
@@ -355,6 +354,24 @@ proc createReservation*(
 
   return success(reservation)
 
+proc changeAvailabilitySize*(
+  self: Reservations,
+  availability: Availability,
+  newSize: UInt256
+  ): Future[?!void] {.async.} =
+  ## Method that adjusts the repo reservations according the changed sizes.
+  ## The Availability.totalSize must not be updated yet!
+
+  if availability.totalSize < newSize: # storage added
+    if reserveErr =? (await self.repo.reserve((newSize - availability.totalSize).truncate(uint))).errorOption:
+      return failure(reserveErr.toErr(ReserveFailedError))
+
+  elif availability.totalSize > newSize: # storage removed
+    if reserveErr =? (await self.repo.release((availability.totalSize - newSize).truncate(uint))).errorOption:
+      return failure(reserveErr.toErr(ReleaseFailedError))
+
+  return success()
+
```

Comment:
Could this be called directly from `update*` and then not exported? It feels more like an "internal operation" of the reservations module.

Reply:
I wanted to do it originally like that, but then I thought I won't have the diff between the old/new values of size, but now I have realized that I can load it from the data store before updating to compare the values like that, so - yes, I will do that!
Thanks ;-)